filecoin-proofs/src/api/update.rs

```diff
@@ -649,3 +714,269 @@ pub fn verify_empty_sector_update_proof<Tree: 'static + MerkleTreeTrait<Hasher =
+        "cannot aggregate with empty sector_update_inputs"
+    );
+
+    let config = SectorUpdateConfig::from_porep_config(porep_config);
+    let partitions = usize::from(config.update_partitions);
+    let verifying_key = get_empty_sector_update_verifying_key::<Tree>(porep_config)?;
+
+    let mut proofs: Vec<_> = proofs
+        .iter()
+        .try_fold(Vec::new(), |mut acc, proof| -> Result<_> {
+            acc.extend(
+                MultiProof::new_from_reader(Some(partitions), proof.0.as_slice(), &verifying_key)?
+                    .circuit_proofs,
+            );
+
+            Ok(acc)
+        })?;
+    trace!(
+        "aggregate_sector_update_proofs called with {} inputs for {} proofs",
+        sector_update_inputs.len(),
+        proofs.len(),
+    );
+
+    // Note that the proofs count here is not the same as the input
+    // proofs since the multi proof type takes the partitions into
+    // account
+    let target_proofs_len = get_aggregate_target_len(proofs.len());
+    ensure!(
+        target_proofs_len > 1,
+        "cannot aggregate less than two proofs"
+    );
+    trace!(
+        "aggregate_seal_commit_proofs will pad proofs to target_len {}",
+        target_proofs_len
+    );
+
+    // If we're not at the pow2 target, duplicate the last proof until we are.
+    pad_proofs_to_target(&mut proofs, target_proofs_len)?;
+
+    // Hash all of the commitments into an ordered digest for the aggregate proof method.
+    let hashed_commitments: [u8; 32] = {
+        let mut hasher = Sha256::new();
+        for input in sector_update_inputs.iter() {
+            hasher.update(input.comm_r_old);
+            hasher.update(input.comm_r_new);
+            hasher.update(input.comm_d_new);
+        }
+        hasher.finalize().into()
+    };
+
+    let srs_prover_key = get_stacked_srs_key::<Tree>(porep_config, proofs.len())?;
+    let aggregate_proof = EmptySectorUpdateCompound::<Tree>::aggregate_proofs(
+        &srs_prover_key,
+        &hashed_commitments,
+        proofs.as_slice(),
+        aggregate_version,
+    )?;
+    let mut aggregate_proof_bytes = Vec::new();
+    aggregate_proof.write(&mut aggregate_proof_bytes)?;
+
+    info!("aggregate_empty_sector_update_proofs:finish");
+
+    Ok(aggregate_proof_bytes)
+}
+
+pub fn verify_aggregate_sector_update_proofs<
+    Tree: 'static + MerkleTreeTrait<Hasher = TreeRHasher>,
+>(
+    porep_config: &PoRepConfig,
+    aggregate_proof_bytes: AggregateSnarkProof,
+    inputs: &[SectorUpdateProofInputs],
+    sector_update_inputs: Vec<Vec<Fr>>,
+    aggregate_version: groth16::aggregate::AggregateVersion,
+) -> Result<bool> {
+    info!("verify_aggregate_sector_update_proofs:start");
+
+    info!(
+        "verify_aggregate_sector_update_proofs using API Version {}",
+        porep_config.api_version
+    );
+    ensure!(
+        porep_config.api_version > ApiVersion::V1_1_0,
+        "Empty Sector Update proof aggregation is supported in ApiVersion 1.2.0 or later"
+    );
```

Comment:
Same comment as previously: I personally find `porep_config.api_version >= ApiVersion::V1_2_0` easier to read. Also what if we would introduce a `ApiVersion::V1_1_1`?